File: app/src/androidTest/java/com/ndipatri/solarmonitor/container/TestObjectGraph.java
package com.ndipatri.solarmonitor.container;

import android.content.Context;

import com.ndipatri.solarmonitor.activities.MainActivity;
import com.ndipatri.solarmonitor.MainActivityInstrumentationTest;
import com.ndipatri.solarmonitor.fragments.ConfigurePanelDialogFragment;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules={ServiceModule.class,
                    HardwareModule.class})
public interface TestObjectGraph extends ObjectGraph {
    void inject(ConfigurePanelDialogFragment fragment);
    void inject(MainActivity activity);
    void inject(MainActivityInstrumentationTest test);

    final class Initializer {
        public static TestObjectGraph init(Context context) {
            return DaggerTestObjectGraph.builder()
                    .serviceModule(new ServiceModule(context))
                    .hardwareModule(new HardwareModule(context))
                    .build();
        }
    }
}

File: app/src/androidTest/java/com/ndipatri/solarmonitor/container/TestObjectGraph.kt
package com.ndipatri.solarmonitor.container

import android.content.Context

import com.ndipatri.solarmonitor.activities.MainActivity
import com.ndipatri.solarmonitor.MainActivityInstrumentationTest
import com.ndipatri.solarmonitor.fragments.ConfigDialogFragment

import javax.inject.Singleton

import dagger.Component

@Singleton
@Component(modules = arrayOf(ServiceModule::class, HardwareModule::class))
interface TestObjectGraph : ObjectGraph {
    override fun inject(fragment: ConfigDialogFragment)
    override fun inject(activity: MainActivity)
    fun inject(test: MainActivityInstrumentationTest)

    object Initializer {
        fun init(context: Context): TestObjectGraph {
            return DaggerTestObjectGraph.builder()
                    .serviceModule(ServiceModule(context))
                    .hardwareModule(HardwareModule(context))
                    .build()
        }
    }
}
